workspaces/src/network/result.rs

```diff
@@ -81,6 +86,60 @@ impl CallExecutionDetails {
         };
         base64::decode(result).map_err(Into::into)
     }
+
+    /// Grab all outcomes after the execution of the transaction. This includes outcomes
+    /// from the transaction and all the receipts it generated.
+    pub fn outcomes(&self) -> Vec<&ExecutionOutcome> {
+        let mut outcomes = vec![&self.transaction];
+        outcomes.extend(&self.receipts);
+        outcomes
+    }
+
+    /// Grab all outcomes after the execution of the transaction. This includes outcomes
+    /// only from receipts generated by this transaction.
+    pub fn receipt_outcomes(&self) -> &Vec<ExecutionOutcome> {
+        &self.receipts
+    }
+
+    /// Grab all outcomes that did not succeed the execution of this transaction. This
+    /// will also include the failures from receipts as well.
+    pub fn failures(&self) -> Vec<&ExecutionOutcome> {
+        let mut failures = Vec::new();
+        if matches!(self.transaction.status, ExecutionStatusView::Failure(_)) {
+            failures.push(&self.transaction);
+        }
+        failures.extend(self.receipt_failures());
+        failures
+    }
+
+    /// Just like `failures`, grab only failed receipt outcomes.
+    pub fn receipt_failures(&self) -> Vec<&ExecutionOutcome> {
+        self.receipts
+            .iter()
+            .filter(|receipt| matches!(receipt.status, ExecutionStatusView::Failure(_)))
+            .collect()
+    }
+
+    /// Checks whether the transaction was successful. Returns true if
+    /// the transaction has a status of FinalExecutionStatus::Success.
```

Comment:
```suggestion
    /// the transaction has a status of [FinalExecutionStatus::Success].
```
Prob worth having intra doc links for these (I didn't check if this is resolved, so you might need to fix if this doesn't)